File: Pov_Back/Pov_Bac/src/main/java/com/talini/pov_bac/model/Produto.java
package com.talini.pov_bac.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @JsonProperty("idProduto")
    private int idProduto;

    @JsonProperty("Descricao")
    private String descricao;

    @JsonProperty("CodBarras")
    private long codBarras;

    @JsonProperty("QtdDisponivel")
    private long qtdDisponivel;

    @JsonProperty("Medida")
    private String medida;

    @JsonProperty("Preco")
    private double preco;
}

File: Pov_Front/lib/DataModels/Produto.dart
class Produto {
  final int idProduto;
  final String Descricao;
  final int CodBarras;
  final int QtdDisponivel;
  final String Medida;

  Produto(
      {required this.idProduto,
      required this.Descricao,
      required this.CodBarras,
      required this.QtdDisponivel,
      required this.Medida});

  factory Produto.fromJson(Map<String, dynamic> json) {
    return Produto(
        idProduto: json['idProduto'],
        Descricao: json['Descricao'],
        CodBarras: json['CodBarras'],
        QtdDisponivel: json['QtdDisponivel'],
        Medida: json['Medida']);
  }

  Map<String, dynamic> toJson() {
    return {
      'idProduto': idProduto,
      'Descricao': Descricao,
      'CodBarras': CodBarras,
      'QtdDisponivel': QtdDisponivel,
      'Medida': Medida
    };
  }
}
